src/sage/homology/homology_vector_space_with_basis.py

```diff
@@ -413,6 +482,221 @@ def to_cycle(self):
+              otherwise ``a * self``
+
+            Algorithm: use the action of the Steenrod algebra `A` on
+            cohomology to construct the action on homology. That is,
+            given a right action of `A` on `H^*`,
+
+            .. MATH::
+
+                \phi_L: H^* \otimes A \to H^*
+
+            we define (a la Boardman [Boa1982]_, p. 190)
+
+            .. MATH::
+
+                S'' \phi_L: A \otimes H_* \to H_*
+
+            using the formula
+
+            .. MATH::
+
+                \langle (S'' \phi) (f \otimes a), x \rangle
+                = \langle f, \phi_L (a \otimes x) \rangle,
+
+            for `f \in H_m`, `a \in A^n`, and `x \in
+            H^{m-n}`. Somewhat more succintly, we define the action `f
+            \cdot a` by
+
+            .. MATH::
+
+                (f \cdot a) (x) = f (a \cdot x)
+
+            So given `f` (a.k.a. ``self``) and `a`, we compute `f (a
+            \cdot x)` for all basis elements `x` in `H^{m-n}`,
+            yielding a vector indexed by those basis elements. Since
+            our basis for homology is dual to the basis for
+            cohomology, we can then use the homology basis to convert
+            the vector to an element of `H_{m-n}`.
+
+            This gives a right module structure. To get a left module
+            structure, use the right module structure after applying
+            the antipode to `a`.
+
+            EXAMPLES::
+
+                sage: # needs sage.groups
+                sage: RP5 = simplicial_sets.RealProjectiveSpace(5)
+                sage: H = RP5.homology_with_basis(GF(2))
+                sage: x5 = list(H.basis(5))[0]
+                sage: Sq(1) * x5
+                0
+                sage: Sq(2) * x5
+                h_{3,0}
+                sage: x5 * Sq(2)
+                h_{3,0}
+
+            TESTS::
+
+                sage: # needs sage.groups
+                sage: RP4 = simplicial_sets.RealProjectiveSpace(5)
+                sage: H = RP4.homology_with_basis(GF(2))
+                sage: x4 = H.basis()[4,0]
+                sage: (Sq(1) * Sq(2)) * x4 != 0
+                True
+                sage: (Sq(1) * Sq(2)) * x4 == Sq(1) * (Sq(2) * x4)
+                True
+                sage: x4 * (Sq(2) * Sq(1)) == (x4 * Sq(2)) * Sq(1)
+                True
+            """
+            # Handle field elements first.
+            if a in self.base_ring():
+                return self.map_coefficients(lambda c: c*a)
+            if not self_on_left: # i.e., module element on left
+                a = a.antipode()
+
+            m = self.degree()
+            n = a.degree()
+            if m <= n:
+                return self.parent().zero()
+
+            vec = []
+            for x in sorted(self.parent().dual().basis(m-n)):
+                vec.append(self.eval(a * x))
+            B = list(self.parent().basis(m-n))
+            return self.parent().linear_combination(zip(B, vec))
```

Comment:
```suggestion
            P = self.parent()
            B = list(P.basis(m-n))
            return P._from_dict({b.support()[0]: self.eval(a * x)
                                 for x in sorted(self.parent().dual().basis(m-n))})
```
This should be equivalent and faster.